lib/AST/ASTImporter.cpp

```diff
@@ -2492,9 +2491,8 @@ ExpectedDecl ASTNodeImporter::VisitEnumDecl(EnumDecl *D) {
       if (auto *FoundEnum = dyn_cast<EnumDecl>(FoundDecl)) {
         if (IsStructuralMatch(D, FoundEnum))
           return Importer.MapImported(D, FoundEnum);
+        ConflictingDecls.push_back(FoundDecl);
```

Comment:
Well, yes, that could be a problem, but in other cases it may not.
E.g. think about the case when we import a FunctionTemplateDecl with name `f` and we found a simple FunctionDecl with name `f`. They overload. 
Or in case of a ClassTemplateDecl and CXXRecordDecl, the CXXRecordDecl could be the 'templated' class, so it would be false to report error.
So I think we should report a name conflict error only when we are 100% sure of that.
That is why I think it should be a general pattern to report the error only if the kind is the same.